src/Classes/ImportTab.lua

```diff
@@ -1111,17 +1136,22 @@ function ImportTabClass:ImportSocketedItems(item, socketedItems, slotName)
 	end
 end
 
--- Return the index of the group with the most gems
+-- Return the index of the group with the most DPS
 function ImportTabClass:GuessMainSocketGroup()
-	local largestGroupSize = 0
-	local largestGroupIndex = 1
-	for i, socketGroup in ipairs(self.build.skillsTab.socketGroupList) do
-		if #socketGroup.gemList > largestGroupSize then
-			largestGroupSize = #socketGroup.gemList
-			largestGroupIndex = i
-		end
-	end
-	return largestGroupIndex
+	local bestDps = 0
+    local bestSocketGroup = nil
+    for i, socketGroup in pairs(self.build.skillsTab.socketGroupList) do
+        self.build.mainSocketGroup = i
+        socketGroup.includeInFullDPS = true
+        local mainOutput = self.build.calcsTab.calcs.buildOutput(self.build, "MAIN").player.output
+        socketGroup.includeInFullDPS = false
+        local dps = mainOutput.FullDPS
+        if dps > bestDps then
+            bestDps = dps
+            bestSocketGroup = i
+        end
+    end
+    return bestSocketGroup
```

Comment:
(these should be tab-indented)

Reply:
The "Refined GitHub" extension for FF can show whitespace in PRs, and I try to take a quick peek of most PRs, especially interesting ones.  😃